### DISCUSSION TITLE
Why do not "go build" before "COPY nats-server /bin/nats-server"?

### DISCUSSION THREAD

**Discussion by penghuazhou** [nats-io/nats-server#5990] [Q&A]
Why do not "go build" before "COPY nats-server /bin/nats-server"? 
for example
```
go build
COPY nats-server /bin/nats-server
```
<img width="1215" alt="image" src="https://github.com/user-attachments/assets/0462bbe4-5d98-4e79-9c53-49f55a63b291">

**Comment by ripienaar** [comment]
The build is done using `goreleaser` so we only copy the binary in.

https://github.com/nats-io/nats-server/blob/2c71b59b430f3f1ec80b6b768ea0a32c1b62c554/.github/actions/nightly-release/action.yaml#L33-L48

**Reply by penghuazhou** [reply]
thanks
